lib/full-site-editing/block-templates.php

```diff
@@ -380,47 +380,63 @@ function gutenberg_get_block_template( $id, $template_type = 'wp_template' ) {
 /**
  * Generates a unique slug for templates or template parts.
  *
- * @param string $slug          The resolved slug (post_name).
+ * @param string $_slug         The filtered value of the slug (starts as `null` from apply_filter).
+ * @param string $slug          The original/un-filtered slug (post_name).
  * @param int    $post_ID       Post ID.
  * @param string $post_status   No uniqueness checks are made if the post is still draft or pending.
  * @param string $post_type     Post type.
  * @return string The original, desired slug.
  */
-function gutenberg_filter_wp_template_unique_post_slug( $slug, $post_ID, $post_status, $post_type ) {
-	if ( 'wp_template' !== $post_type || 'wp_template_part' !== $post_type ) {
-		return $slug;
+function gutenberg_filter_wp_template_unique_post_slug( $_slug, $slug, $post_ID, $post_status, $post_type ) {
+	if ( 'wp_template' !== $post_type && 'wp_template_part' !== $post_type ) {
+		return $_slug;
+	}
+
+	if ( ! $_slug ) {
+		$_slug = $slug;
```

Comment:
This is a bit hard to read, and also to follow in the code below. Can we use the default parameter name from core at least? (`$override_slug`, [reference](https://developer.wordpress.org/reference/hooks/pre_wp_unique_post_slug/)). Also the parameter description from that link could be useful for the function doc.

Reply:
updated